modules/internal/String.chpl

```diff
@@ -500,12 +500,15 @@ module String {
     return x != 0;
   // End index arithmetic support
 
-  private proc validateEncoding(buf, len) throws {
-    extern proc chpl_enc_validate_buf(buf, len) : c_int;
-
-    if chpl_enc_validate_buf(buf, len) != 0 {
+  private proc validateEncoding(buf, len): int throws {
+    var numCodepoints: int;
+    extern proc chpl_enc_validate_buf(buf, len, ref numCodepoints) : c_int;
```

Comment:
Can you put the extern declaration above `numCodepoints`'s definition, and ideally add a newline in between the two